index.bs

```diff
@@ -225,7 +225,17 @@ its \[[Prototype]] an empty object, created as if by ObjectCreate(%ObjectPrototy
 
 <h4 id="assert" oldids="assert-condition-data,dom-console-assert" method for="console">assert(<var>condition</var>, ...<var>data</var>)</h4>
 
-If <var>condition</var> is false, perform Logger("error", <var>data</var>).
+<emu-alg>
+  1. If _condition_ is true, abort these steps.
+  1. Let _message_ be a string without any formatting specifiers indicating generically an assertion failure (such as "Assertion failed").
+  1. If _data_ is empty, append _message_ to _data_.
+  1. Otherwise, implementations should perform these substeps:
+    1. Let _first_ be the first element of _data_.
+    1. If _first_ is not a string, prepend _message_ to _data_. Abort these substeps.
```

Comment:
"is not a string" is kind of ambiguous. What do browsers do if it is `{ toString() { "some message" } }`? Do they convert it to a string?

If we want a type test here, this should be "If Type(_first_) is String, ...". But I am not sure that's what we want...